src/config/config.rs

```diff
@@ -17,37 +17,52 @@ use std::{
     str::FromStr,
 };
 
-fn canonicalize(path: &Path) -> String {
-    let mut parts: Vec<String> = Vec::new();
+fn relative_canonicalize(path: impl AsRef<Path>) -> PathBuf {
+    let mut parts: Vec<Component<'_>> = vec![];
 
-    for component in path.components() {
+    for component in path.as_ref().components() {
         match component {
-            Component::RootDir | Component::CurDir => {}
+            // :shrug:
+            // Component::RootDir | Component::CurDir => {}
```

Comment:
`CurDir` is basically ".", there is no point into keeping it.

Reply:
Fair, `foo/./bar` should just become `foo/bar`. I would like to preserve `RootDir` though, for correctness (even if it is unlikely to be used).